### PR TITLE
feat(amplify_auth_cognito): enables clientMetadata for signUp

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
 Enables clientMetadata for signUp

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift [hunk 1]
```diff
@@ -281,14 +282,34 @@ class amplify_auth_cognito_tests: XCTestCase {
         XCTAssertNoThrow(try FlutterSignUpRequest.validate(dict: rawData))
     }
     
-    func test_signUpFormatAttributes() {
+    func test_signUpFormatAttributesWithUserAttributes() {
```
Comment (Jordan-Nelson) [inline packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift:285]: *nit:* The name on each of these could probably be updated to remove the `FormatAttributes`
```suggestion
    func test_signUpWithUserAttributes() {
```

File: packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift [hunk 2]
```diff
@@ -281,14 +282,34 @@ class amplify_auth_cognito_tests: XCTestCase {
...
         let rawAttributes: Dictionary<String, Any> = ["email": _email, "customAttribute": "female"]
         let rawOptions: Dictionary<String, Any> = ["userAttributes": rawAttributes]
         let rawData: NSMutableDictionary = ["options":rawOptions, "username": _username, "password": _password]
         let request = FlutterSignUpRequest(dict: rawData);
         XCTAssertEqual(2, request.options?.userAttributes?.count)
     }
     
+    func test_signUpFormatAttributesWithClientMetadata() {
```
Comment (Jordan-Nelson) [inline packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift:293]:
```suggestion
    func test_signUpWithClientMetadata() {
```

File: packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift [hunk 3]
```diff
@@ -281,14 +282,34 @@ class amplify_auth_cognito_tests: XCTestCase {
...
+        let metadata: Dictionary<String, Any> = ["attribute": "value"]
+        let rawOptions: Dictionary<String, Any> = ["clientMetadata": metadata]
+        let rawData: NSMutableDictionary = ["options":rawOptions, "username": _username, "password": _password]
+        let request = FlutterSignUpRequest(dict: rawData);
+        let options = request.options?.pluginOptions as! AWSAuthSignUpOptions
+        XCTAssertEqual("value", options.metadata!["attribute"])
+        XCTAssertNil(options.validationData)
+    }
+    
+    func test_signUpFormatAttributesWithValidationdata() {
```
Comment (Jordan-Nelson) [inline packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift:303]:
```suggestion
    func test_signUpWithValidationdata() {
```

**Review by Jordan-Nelson — APPROVED** [review]
A few nit comments on naming in tests, but LGTM otherwise

**Review by offlineprogrammer — APPROVED** [review]

**Review by dnys1 — APPROVED** [review]

**Pull request merged**